File: python/problems/leetcode/1704-Determine-if-String-Halves-Are-Alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = {"a", "A", "e", "E", "i", "I", "o", "O", "u", "U"}
        first = 0
        second = 0

        left, right = 0, len(s) - 1

        while left <= right:
            if s[left] in vowels:
                first += 1
            if s[right] in vowels:
                second += 1
            left += 1
            right -= 1
        return first == second

File: cpp/problems/leetcode/1704-Determine-if-String-Halves-Are-Alike.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
 public:
  bool halvesAreAlike(string s) {
    set<char> vowels = {'a', 'A', 'e', 'E', 'i', 'I', 'o', 'O', 'u', 'U'};
    int left = 0, right = s.size() - 1;
    int first = 0, second = 0;
    while (left <= right) {
      if (vowels.find(s[left]) != vowels.end()) {
        first++;
      }
      if (vowels.find(s[right]) != vowels.end()) {
        second++;
      }
      left++;
      right--;
    }

    return first == second;
  }
};